src/Http/TraceIdAwareHttpClient.php

```diff
@@ -15,9 +15,9 @@
 class TraceIdAwareHttpClient implements HttpClientInterface, ResetInterface, LoggerAwareInterface
```

Comment:
This class would then be obsolete. We would just swap the TraceContextService.

Reply:
Yes, still working on the PR ;) (still a draft).
Wanted to start with a base setup and have the tests ready and passing.